nifi-bootstrap/src/main/java/org/apache/nifi/bootstrap/RunNiFi.java

```diff
@@ -1165,13 +1171,22 @@ public void start(final boolean monitor) throws IOException {
 
         nifiPropsFilename = nifiPropsFilename.trim();
 
+        String xmx = null;
+        String xms = null;
+
         final List<String> javaAdditionalArgs = new ArrayList<>();
         for (final Map.Entry<String, String> entry : props.entrySet()) {
             final String key = entry.getKey();
             final String value = entry.getValue();
 
             if (key.startsWith("java.arg")) {
                 javaAdditionalArgs.add(value);
+                if(value.toLowerCase().startsWith("-xms")) {
+                    xms = StringUtils.substringAfter(value.toLowerCase(), "-xms");
+                }
+                if(value.toLowerCase().startsWith("-xmx")) {
+                    xmx = StringUtils.substringAfter(value.toLowerCase(), "-xmx");
+                }
```

Comment:
Instead of parsing `Xmx`, what about using [Runtime.maxMemory()](https://docs.oracle.com/javase/8/docs/api/java/lang/Runtime.html#maxMemory--)?

Reply:
That would be the value for the bootstrap process (48MB), not for the NiFi JVM

Reply:
That's a good point, thanks!